app/src/main/java/net/bible/service/db/DatabaseContainer.kt

```diff
@@ -71,17 +71,19 @@ private val MIGRATION_5_6 = object : Migration(5, 6) {
 private val MIGRATION_6_7 = object : Migration(6, 7) {
     override fun migrate(db: SupportSQLiteDatabase) {
         db.apply {
+            execSQL("PRAGMA foreign_keys=ON;")
             execSQL("DROP TRIGGER IF EXISTS bookmark_cleanup;")
             execSQL("DROP TRIGGER IF EXISTS label_cleanup;")
             execSQL("CREATE TABLE `bookmark_label_new` (`bookmark_id` INTEGER NOT NULL, `label_id` INTEGER NOT NULL, PRIMARY KEY(`bookmark_id`, `label_id`), FOREIGN KEY(`bookmark_id`) REFERENCES `bookmark`(`_id`) ON UPDATE NO ACTION ON DELETE CASCADE , FOREIGN KEY(`label_id`) REFERENCES `label`(`_id`) ON UPDATE NO ACTION ON DELETE CASCADE );");
-            execSQL("CREATE TABLE `readingplan_status_new` (`_id` INTEGER, `plan_code` TEXT NOT NULL, `plan_day` INTEGER NOT NULL, `reading_status` TEXT NOT NULL, PRIMARY KEY(`_id`), FOREIGN KEY(`plan_code`) REFERENCES `readingplan`(`_id`) ON UPDATE NO ACTION ON DELETE CASCADE );");
+            execSQL("CREATE TABLE `readingplan_status_new` (`_id` INTEGER, `plan_code` TEXT NOT NULL, `plan_day` INTEGER NOT NULL, `reading_status` TEXT NOT NULL, PRIMARY KEY(`_id`), FOREIGN KEY(`plan_code`) REFERENCES `readingplan`(`plan_code`) ON UPDATE NO ACTION ON DELETE CASCADE )")
```

Comment:
I am not sure if it is a good idea to have a foreign key with plan_code. Will child table be able to have a record without parent table having that plan_code?

As I said in my other comment, when user starts a plan, reading_status gets records added as soon as a chapter is marked as read, but readingplan only gets record added when user goes to second day. I'm not sure if that will be allowed with the foreign key?

Reply:
Okay, dropped foreign key from here.

Reply:
I think you missed line 78?

Reply:
True, fixed